### ISSUE TITLE
Periodic Duchon unusable: gaussian_reml_fit_positions rejects every period (phantom 0.875 support range; period=None non-PSD)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#580]
## Summary

`gaussian_reml_fit_positions(..., periodic=True, basis="duchon")` **cannot be
called successfully by any combination of inputs**. Every `period` value is
rejected against an internally-computed "support range" that equals neither the
data span nor any period the caller can pass, and `period=None` produces a
non-PSD penalty. This makes cyclic/periodic Duchon curves unreachable through
the public API.

## Repro

```python
import numpy as np, torch, gamfit
torch.set_default_dtype(torch.float64)
rng = np.random.default_rng(1)
N = 600
t = np.linspace(0, 1, N, endpoint=False)          # canonical [0,1) periodic grid
th = 2*np.pi*t
Q, _ = np.linalg.qr(rng.standard_normal((16, 2)))
y = torch.tensor(np.stack([np.cos(th), np.sin(th)], -1) @ Q.T)   # clean circle in R^16
tt = torch.tensor(t)

span = float(t.max() - t.min())                   # 0.9983333...
for period in (span, 1.0, None):
    try:
        gamfit.gaussian_reml_fit_positions(tt, y, basis="duchon", basis_order=2,
                                           periodic=True, period=period)
        print(period, "OK")
    except Exception as e:
        print(period, "FAIL:", str(e).splitlines()[0])
```

Output:
```
0.9983333333333334  FAIL: duchon periodic support range (0.8750000000000001) must match explicit period (0.9983333333333334)
1.0                 FAIL: duchon periodic support range (0.8750000000000001) must match explicit period (1)
None                FAIL: Invalid input: Gaussian REML penalty is not positive semidefinite; eigenvalue=-6.250e2
```

## Two distinct bugs

1. **Phantom support range `0.875`.** With explicit `period`, the guard compares
   against a support range of `0.8750000000000001` (= 7/8) that matches neither
   the data span (`0.9983`) nor the requested period. `7/8` looks knot-derived,
   not data-derived — the support-range computation for the periodic Duchon
   guard appears to use an internal knot extent rather than the actual `t`
   range, so the equality check can never pass for real data on `[0, 1)`.
2. **`period=None` → non-PSD penalty.** The documented auto path builds a
   Gaussian REML penalty with a large negative eigenvalue (`-625`) and is
   rejected by the PSD check, so auto-period periodic Duchon also can't run.

## Impact

Periodic atoms are a stated Manifold-SAE feature ("gamfit has periodic-Duchon").
With both the explicit and auto paths broken, closed curves (circles, cyclic
features like day-of-week/hue) can't be fit with periodic Duchon at all. The
non-periodic `gaussian_reml_fit_positions` works fine (R²≈0.996 on an open
curve), and the dedicated `periodic_spline_curve_basis` + `gaussian_weighted_ridge`
recovers a circle at R²≈0.998 — so the gap is specifically periodic **Duchon**.

## Ask
- Compute the periodic support range from the actual `t` extent (or document the
  exact convention) so an explicit `period` matching the data span is accepted.
- Fix the `period=None` penalty construction so it is PSD.

## Environment
gamfit 0.1.141, torch 2.12.0, CPU, macOS arm64.

**Comment by SauersML** [comment]
## Update — partial fix landed, deeper root cause isolated

### Fixed (validation + period plumbing)
The user's `period` was used **only** by `validate_position_period` and never
reached the basis or penalty builder — both passed `None` to
`build_duchon_basis_mixed_periodicity_auto`, which auto-derived the period as the
center span. And the guard required `span == period`, so every explicit period
was rejected. Fixed:
- `validate_position_period`: the period is the domain **wrap**, not the sample
  span; require `period >= span` (centers fit within one period), not `==`.
- Thread the explicit 1D `period` into both `duchon_function_norm_penalty` and
  `duchon_basis_1d_impl` (→ `build_duchon_basis_mixed_periodicity_auto(..,
  Some(&[period]))`).
- Python `_resolve_position_penalty` / `_resolve_position_basis_inputs` now pass
  the period through (was hardcoded `period=None`) and auto-derive
  `span + one_spacing` when periodic and no period is given, so basis and penalty
  share the same wrap.

After this, `period=1.0` is **accepted** (the "support range must match" error is
gone) and the period flows consistently. No regression: non-periodic forward
R²=0.998, batched path fine.

### Deeper root cause that remains: the periodic Gram is genuinely indefinite
With the period correct, the penalty is **still non-PSD** — and it is not
numerical noise or the period. The periodic Duchon function-norm Gram (10×10,
m=2, period=1.0) spectrum:

```
eig = [-0.700, -0.700, 0, 0.0024, 0.0036, 0.0036, 0.0114, 0.0114, 0.100, 0.100]
```

A **degenerate −0.70 pair dominates** the max positive eigenvalue (0.10). The
non-periodic Duchon penalty on the same knots is cleanly PSD (min eig 0.0).

The chord-distance polyharmonic kernel `r^{2m−d} = r³` is **conditionally** PD on
ℝ but is **not** conditionally PD on the circle under the chord metric, so the
periodic function-norm Gram is indefinite. The non-periodic path lands PSD
because it applies the polynomial-nullspace conditioning/projection; the
mixed-periodicity Gram builder does not produce a PSD penalty on the penalized
subspace.

### Ask / next step
The periodic position penalty needs to be a genuinely PSD object — either apply
the conditional-PD nullspace projection in `build_duchon_basis_mixed_periodicity`
for the periodic case, or route periodic 1D position fits to a PSD periodic
penalty (cyclic difference / Fourier). Until then the working path for closed
curves is `periodic_spline_curve_basis` + `gaussian_weighted_ridge` (R²≈0.998 on
a circle in R¹⁶).

Env: gamfit 0.1.141 (local build), torch 2.12, CPU, macOS arm64.

**Comment by SauersML** [comment]
🔒 claim: claude-code coordinator session — taking this now. (cluster: gaussian-reml-positions)
